woocommerce/woocommerce
# Add Playwright-Allure integration

### Changes proposed in this Pull Request:

Closes 228-gh-woocommerce/woocommerce-quality

This PR is intended to be merged with branch `add/playwright-e2e` only, not `trunk`.

This PR just adds the dependencies needed to integrate Allure with our Playwright tests.

### How to test the changes in this Pull Request:

1. Clone this repo and switch to this branch.
2. Launch the e2e testing environment on your local using the steps in [Prep work for running tests](https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce/tests/e2e#prep-work-for-running-tests):
3. `cd` into `plugins/woocommerce`.
4. Run the playwright tests and wait for them to finish: `pnpx playwright test --config=e2e/playwright.config.js`
5. Verify that the folder `plugins/woocommerce/e2e/allure-results` was automatically created at the end of the test run.
6. Verify that this folder contains a bunch of .json files.
7. Still in the `plugins/woocommerce` folder, run the command to generate the Allure report: `pnpx allure generate --clean e2e/allure-results --output e2e/allure-report`
8. Verify that the folder `plugins/woocommerce/e2e/allure-report` was created, and contains the static files for the Allure report.
9. To open the Allure report, run `pnpx allure open e2e/allure-report`



### FOR PR REVIEWER ONLY:
